### PR TITLE
Add coverage report and badge

### LINKED ISSUE #18: Add coverage reports to jest
Use the `--coverage` option in jest to collect and report code coverage

Consider adding a badge to our README

### PR DESCRIPTION
**Pull request by bmind12**
Solves #18 

Adds two new scripts: `test:coverage` and `test:badge`. `test:badge` - generates a coverage badge and saves it as `assets/badge.svg`

Adds the generated badge to `README.md`

Adds `test:badge` script into github workflow to generate a new badge

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - run: npm ci
-    - run: npm test
+    - run: npm test:badge
```
Comment (dangoslen) [inline .github/workflows/test.yml]: I think this now needs to be `npm run test:badge` right?

File: README.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
   <h2>Keeping repositories contributable.</h2>
 
   <a href="https://github.com/dangoslen/constable-github-action/actions"><img alt="constable-github-action-status" src="https://github.com/dangoslen/constable-github-action/workflows/units-test/badge.svg"></a>
+  <img alt="constable-github-action-status" src="./assets/badge.svg">
```
Comment (dangoslen) [inline README.md]: Consider making the `alt=constable-github-action-coverage` as I believe it is clearer

File: assets/badge.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="106" height="20" role="img" aria-label="Coverage: 100%"><title>Coverage: 100%</title><linearGradient id="s" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="r"><rect width="106" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#r)"><rect width="63" height="20" fill="#555"/><rect x="63" width="43" height="20" fill="#4c1"/><rect width="106" height="20" fill="url(#s)"/></g><g fill="#fff" text-anchor="middle" font-family="Verdana,Geneva,DejaVu Sans,sans-serif" text-rendering="geometricPrecision" font-size="110"><text aria-hidden="true" x="325" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="530">Coverage</text><text x="325" y="140" transform="scale(.1)" fill="#fff" textLength="530">Coverage</text><text aria-hidden="true" x="835" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="330">100%</text><text x="835" y="140" transform="scale(.1)" fill="#fff" textLength="330">100%</text></g></svg>
```
Comment (dangoslen) [inline assets/badge.svg]: could we name this `assests/coverage-badge.svg`? That way, if we create additional badges we wouldn't get any named conflicts.

**Review by dangoslen — COMMENTED** [review]
Thanks for contributing!! I have a few small things, but overall it looks great.

**Comment by bmind12** [comment]
> Thanks for contributing!! I have a few small things, but overall it looks great.

Thanks for the feedback! I did the changes, please take another look.

**Review by dangoslen — APPROVED** [review]
Looks great!

I'll likely merge this tomorrow if that is ok

**Comment by dangoslen** [comment]
Merging now. 🚀 

Thanks again!

**Pull request merged**